pkg/api/helpers_test.go

```diff
@@ -297,3 +297,74 @@ func TestGetAffinityFromPod(t *testing.T) {
 		}
 	}
 }
+
+func TestGetAvoidPodsFromNode(t *testing.T) {
+	testCases := []struct {
+		node      *Node
+		expectErr bool
+	}{
+		{
+			node:      &Node{},
+			expectErr: false,
+		},
+		{
+			node: &Node{
+				ObjectMeta: ObjectMeta{
+					Annotations: map[string]string{
+						PreferAvoidPodsAnnotationKey: `
+							{
+							    "preferAvoidPods": [
+							        {
+							            "podSignature": {
+							                "podController": {
+							                    "kind": "ReplicationController",
+							                    "namespace": "default",
+							                    "name": "foo",
+							                    "apiVersion": "v1"
+							                }
+							            },
+							            "reason": "some reason",
+							            "message": "some message"
+							        }
+							    ]
+							}`,
+					},
+				},
+			},
+			expectErr: false,
+		},
+		{
+			node: &Node{
+				ObjectMeta: ObjectMeta{
+					Annotations: map[string]string{
+						PreferAvoidPodsAnnotationKey: `
+							{
+							    "preferAvoidPods": [
+							        {
+							            "podSignature": {
+							                "podController": {
+							                    "kind": "ReplicationController",
+							                    "namespace": "default",
+							                    "apiVersion": "v1"
+							                }
+							            },
+							            "reason": "some reason",
+							            "message": "some message"
+							}`,
+					},
+				},
+			},
+			expectErr: true,
+		},
+	}
+
+	for i, tc := range testCases {
+		_, err := GetAvoidPodsFromNodeAnnotations(tc.node.Annotations)
```

Comment:
Can you also check the value you got back (in the case where err == nil) to make sure the fields are what you expect?